### PR TITLE
feat(java): add third parties Google BigQuery rule (CWE-201)

### PR DESCRIPTION
**Pull request by elsapet**
## Description

Add Java third party rule for Google BigData

Relates to #197

## Checklist

- [ ] I've added a snapshot that shows my rule works as expected.
- [ ] My rule has adequate metadata to explain its use.
- [ ] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) format

### CODE REVIEW

File: rules/java/third_parties/bigquery.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      - variable: BUILDER
+        detection: java_third_parties_bigquery_builder
+      - variable: HASH_MAP
+        detection: java_shared_lang_instance
+        scope: cursor
+        filters:
+          - variable: JAVA_SHARED_LANG_INSTANCE_TYPE
+            regex: \A(java\.util\.)?HashMap\<\>\z
+trigger:
+  required_detection: java_third_parties_bigquery_row_with_data
```
Comment (elsapet) [inline rules/java/third_parties/bigquery.yml:16]: We need to match on a HashMap that has contains sensitive data. Not sure how to catch a case like the below, for example, while excluding plain old `new HashMap<>();`

```java
Map<String, String> rowContent = new HashMap<>();
rowContent.put("name", user.name);
```

Comment (elsapet) [reply rules/java/third_parties/bigquery.yml:16]: @cfabianski could we discuss this - when you are free :)

Comment (elsapet) [reply rules/java/third_parties/bigquery.yml:16]: Similar case with Segment - https://github.com/Bearer/bearer-rules/pull/267

Comment (didroe) [reply rules/java/third_parties/bigquery.yml:16]: This isn't possible at the moment. I think what you have is the best approach for now.

**Pull request closed without merging**